storage/remote/client.go

```diff
@@ -375,7 +375,8 @@ func (c *Client) Read(ctx context.Context, query *prompb.Query, sortSeries bool)
 	httpReq.Header.Set("User-Agent", UserAgent)
 	httpReq.Header.Set("X-Prometheus-Remote-Read-Version", "0.1.0")
 
-	ctx, cancel := context.WithTimeout(ctx, c.timeout)
+	errTimeout := fmt.Errorf("request timed out after %s", c.timeout)
```

Comment:
Please add a **unit test** to show that this achieves the desired result - as in the cause text is actually returned for someone using Prometheus.

Also for backwards compatibility the error should still contain the original error text as a sub-string/prefix - just in case someone has a log based alert on it or something.